stoppable_workers.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"context"
+	"sync"
+)
+
+// StoppableWorkers is a collection of goroutines that can be stopped at a later time.
+type StoppableWorkers interface {
+	AddWorkers(...func(context.Context))
+	Stop()
+	Context() context.Context
+}
+
+// stoppableWorkersImpl is the implementation of StoppableWorkers. The linter will complain if you
+// try to make a copy of something that contains a sync.WaitGroup (and returning a value at the end
+// of NewStoppableWorkers() would make a copy of it), so we do everything through the
+// StoppableWorkers interface to avoid making copies (since interfaces do everything by pointer).
+type stoppableWorkersImpl struct {
+	mu                      sync.Mutex
+	cancelCtx               context.Context
+	cancelFunc              func()
+	activeBackgroundWorkers sync.WaitGroup
+}
+
+// NewStoppableWorkers runs the functions in separate goroutines. They can be stopped later.
+func NewStoppableWorkers(funcs ...func(context.Context)) StoppableWorkers {
```

Comment:
I changed the design here a bit and did not accept any workers in the constructor. I also changed `AddWorkers` to `Add` and had it accept only a single worker. I'd rather the pattern be that developers create a `StoppableWorkers` instance they store on _their_ struct, and then add workers _one at a time_. Adding multiple workers at once opens us up to more questions like "what if a `Stop` comes in between the addition of the second and third worker?" NetCode has found this logic to be exceedingly fragile, so trying to stay as simple as possible here.